Build is broken with rustc nightly 1.33.0

The build is broken on the latest Rust nightly
```
> rustup run nightly rustc --version
rustc 1.33.0-nightly (bf669d1e3 2019-01-25)
> cargo +nightly test
   Compiling downcast v0.9.2 (/usr/home/somers/src/rust/downcast-rs)
error[E0034]: multiple applicable items in scope
   --> src/lib.rs:120:38
    |
120 |     fn is_type(&self) -> bool { self.type_id() == TypeId::of::<T>() }
    |                                      ^^^^^^^ multiple `type_id` found
    |
note: candidate #1 is defined in the trait `Any`
   --> src/lib.rs:29:5
    |
29  |     fn type_id(&self) -> TypeId { TypeId::of::<Self>() }
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^
    = help: to disambiguate the method call, write `Any::type_id(&self)` instead
note: candidate #2 is defined in the trait `std::any::Any`
    = help: to disambiguate the method call, write `std::any::Any::type_id(&self)` instead

error[E0034]: multiple applicable items in scope
   --> src/lib.rs:120:38
    |
120 |     fn is_type(&self) -> bool { self.type_id() == TypeId::of::<T>() }
    |                                      ^^^^^^^ multiple `type_id` found
    |
note: candidate #1 is defined in the trait `Any`
   --> src/lib.rs:29:5
    |
29  |     fn type_id(&self) -> TypeId { TypeId::of::<Self>() }
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^
    = help: to disambiguate the method call, write `Any::type_id(&self)` instead
note: candidate #2 is defined in the trait `std::any::Any`
    = help: to disambiguate the method call, write `std::any::Any::type_id(&self)` instead

error: aborting due to previous error

For more information about this error, try `rustc --explain E0034`.
error: aborting due to previous error

For more information about this error, try `rustc --explain E0034`.
error: Could not compile `downcast`.
warning: build failed, waiting for other jobs to finish...
error: Could not compile `downcast`.

To learn more, run the command again with --verbose.
```

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -25,6 +25,10 @@ fn type_name<T: StdAny + ?Sized>() -> &'static str { unsafe { intrinsics::type_n
 fn type_name<T: StdAny + ?Sized>() -> &'static str { "[ONLY ON NIGHTLY]" }
 
 pub trait Any: StdAny {
+    /// TODO: once 1.33.0 is the minimum supported compiler version, remove
+    /// Any::type_id_compat and use StdAny::type_id instead.
+    /// https://github.com/rust-lang/rust/issues/27745
+    fn type_id_compat(&self) -> TypeId { TypeId::of::<Self>() }
     #[doc(hidden)]
     fn type_name(&self) -> &'static str { type_name::<Self>() }
 }
@@ -115,7 +119,7 @@ fn to_trait_object<T: ?Sized>(obj: &T) -> TraitObject {
 pub trait Downcast<T>: Any
     where T: Any
 {
-    fn is_type(&self) -> bool { self.type_id() == TypeId::of::<T>() }
+    fn is_type(&self) -> bool { self.type_id_compat() == TypeId::of::<T>() }
 
     unsafe fn downcast_ref_unchecked(&self) -> &T { &*(to_trait_object(self).data as *mut T) }
 

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -24,9 +24,7 @@ fn type_name<T: StdAny + ?Sized>() -> &'static str { unsafe { intrinsics::type_n
 #[cfg(not(feature = "nightly"))]
 fn type_name<T: StdAny + ?Sized>() -> &'static str { "[ONLY ON NIGHTLY]" }
 
-/// FIXME(https://github.com/rust-lang/rust/issues/27745) remove this
 pub trait Any: StdAny {
-    fn type_id(&self) -> TypeId { TypeId::of::<Self>() }
     #[doc(hidden)]
     fn type_name(&self) -> &'static str { type_name::<Self>() }
 }

